### PR TITLE
Implement wc

### PR DESCRIPTION
**Pull request by silverweed**
Addresses #6.

Main differences from GNU `wc`:
- missing `--files0-from` flag
- output numbers are always separated by a single space, rather than spaced in fancy ways.

I included a single unit test, but more may be added with ease if needed.

### CODE REVIEW

**Comment by silverweed** [comment]
I just realized I forgot to add the "total" line. Gonna add it momentarily.

**Comment by GrayJack** [comment]
Nice, I'll start to review in a moment :smile:

File: wc/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+                total_result = total_result.combine(result);
+            }
+        }
+    }
+
+    if filenames.len() > 1 {
+        print_result("total", &total_result);
+    }
+
+    process::exit(0);
```
Comment (GrayJack) [inline wc/src/main.rs]: `process::exit()` do not drop what needs to be dropped, so when it's finishing successfully it's a better pattern to just `return` in early ends, or in this case just let the main finish and drop what needs to be dropped, while if needs other exit codes, drop everything manually and `process::exit()`

File: wc/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+        result.bytes += line.len() as u64 + 1;
+    }
+
+    Ok(result)
+}
+
+fn print_result(filename: &str, result: &WcResult) {
+    println!("{}", get_formatted_result(filename, result));
+}
```
Comment (GrayJack) [inline wc/src/main.rs]: I think this part is small enough to be in main

File: wc/src/wc.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+name: echo
+version: "0.0.0"
+author: Giacomo Parolini <silverweed1991@gmail.com>
+about: "Print newline, word and byte counts for each FILE, and a total line if more than one FILE is specified. A word is a non-zero-length sequence of characters delimited by white space.\n\nWith no FILE, or when FILE is -, read standard input."
```
Comment (GrayJack) [inline wc/src/wc.yml]: Use the word `display` over `print`

File: wc/src/wc.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+args:
+    - FILE:
+        help: One or more file names.
+        required: false 
+        multiple: true
+    - bytes:
+        long: bytes
+        short: c
+        help: print the byte counts
```
Comment (GrayJack) [inline wc/src/wc.yml]: First letter always upper case plus the display-over-print

**Comment by silverweed** [comment]
Thanks for the review! 
I think I'm not gonna implement `--files0-from` at this time, but I'm definitely implementing `--print / -p` :)

File: wc/src/wc.yml (hunk 3 above)
Comment (silverweed) [reply wc/src/wc.yml]: Ok. Just in the "about" or also in all the flags help section?

**Comment by GrayJack** [comment]
Nice, I'm looking forward to this :smiley:

**Comment by GrayJack** [comment]
Looks great, if you fell like that don't need any changes, let me know so I merge it :smile:

**Comment by silverweed** [comment]
It's all good for me :)

**Pull request merged**